ddl/table.go

```diff
@@ -960,6 +961,12 @@ func updateVersionAndTableInfoWithCheck(t *meta.Meta, job *model.Job, tblInfo *m
 // updateVersionAndTableInfo updates the schema version and the table information.
 func updateVersionAndTableInfo(t *meta.Meta, job *model.Job, tblInfo *model.TableInfo, shouldUpdateVer bool) (
 	ver int64, err error) {
+	failpoint.Inject("mockUpdateVersionAndTableInfoErr", func(val failpoint.Value) {
+		if val.(bool) {
+			job.State = model.JobStateCancelled
```

Comment:
This function doesn't update `job.State`, so why we set `job.State` in failpoint.

Reply:
With the mock error, the job will always fail in stateNone until 512 retries. Making it cancelled can accerlate finishing the job, because all we want is to test the args correctness.

Reply:
We can set the retry count to 1(`variable.SetDDLErrorCountLimit(1)`).

Reply:
ok

Reply:
Change it error count limit will mean setting the job to cancelling state which will override the origin error message.
In my test
`failpoint.Return(ver, errors.New("mock update version and tableInfo error,jobID="+strconv.Itoa(int(job.ID))))`
the origin error contains the job ID, when ddl finished, we can directly get the job id from err.Error() and fetch the job from history